### PR TITLE
LVBAG: Convert MultiPolygon to Polygon

### PR DESCRIPTION
**Pull request by yorickdewid**
## What are related issues/pull requests?

Addresses issue #3581

## Tasklist

 - [x] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by rouault** [comment]
The failures in gcore/transformer.py are due to a PROJ master change. Likely a bug. I'm investigating.

File: autotest/ogr/ogr_lvbag.py [hunk 1]
```diff
@@ -342,6 +342,34 @@ def test_ogr_lvbag_invalid_polygon():
     feat = lyr.GetNextFeature()
     assert feat is None
 
+def test_ogr_lvbag_fix_invalid_polygon_to_polygon():
+
+    pytest.skip()
```
Comment (rouault) [inline autotest/ogr/ogr_lvbag.py:347]: why is this disabled ? Same for test_ogr_lvbag_fix_invalid_polygon

**Comment by rouault** [comment]
> The failures in gcore/transformer.py are due to a PROJ master change. Likely a bug. I'm investigating.

fixed

**Comment by yorickdewid** [comment]
I suppose they need to be re-enabled when the PROJ bug is resolved.

**Comment by justb4** [comment]
Thanks for picking up! 
Possible refinement: in some cases GEOS `makeValid()` creates a `GeometryCollection` i.s.o. a `MultiPolygon`. 
So around [ogrlvbaglayer.cpp:705](https://github.com/OSGeo/gdal/blob/9f4a8f51a4d8f64d0ae55303ddfc7d1c6cced5e9/gdal/ogr/ogrsf_frmts/lvbag/ogrlvbaglayer.cpp#L705): add the condition `&& poGeom->getGeometryType() == wkbGeometryCollection` and cast to `toGeometryCollection() ` and `getGeometryType() == wkbPolygon` to get the largest `Polygon`. 
Sorry don't have a specific object id for that case.

**Comment by justb4** [comment]
Note that most if not all invalid Geometries are History records, basically deleted records, but nothing is ever deleted from this dataset. Almost all applications ignore history records unless for historic analysis. 

Another approach is to not try to repair but add a boolean attribute `geom_valid=false`. For 18 million (about half are History) Building-records saves quite some processing and it becomes easier to report real errors back to Kadaster via "terugmeldvoorziening" in [BAGViewer](https://bagviewer.kadaster.nl).

**Comment by yorickdewid** [comment]
@justb4 I've not run this code with a geometry collection, however a base `OGRGeometry` should survive a downcast to its own `OGRGeometryCollection` type.

File: autotest/ogr/ogr_lvbag.py (hunk 1 above)
Comment (rouault) [reply autotest/ogr/ogr_lvbag.py:347]: the pytest.skip() is still there

**Comment by rouault** [comment]
ah, you probably need to test that MakeValid works and pytest.skip() if not.
Like do
```python
    g = ogr.CreateGeometryFromWkt('POLYGON ((0 0,1 1,0 1,1 0,0 0))')
    if g.MakeValid() is None:
        pytest.skip("MakeValid() not available")
```

**Comment by rouault** [comment]
autotest/ogr/ogr_lvbag.py:36:1: F401 'ogrtest' imported but unused

**Pull request merged**